src/main/java/com/leverx/springapp/demo/controller/UserController.java

```diff
@@ -0,0 +1,52 @@
+package com.leverx.springapp.demo.controller;
+
+import com.leverx.springapp.demo.model.User;
+import com.leverx.springapp.demo.repository.UserRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private UserRepository userRepository;
+
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @PostMapping
+    public ResponseEntity createUser(@Valid @RequestBody User user) {
+        userRepository.save(user);
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest().path("/{id}")
+                .buildAndExpand(user.getId()).toUri();
```

Comment:
@blackavgn despite the fact, that user object will receive id after save method, theoretically this code might stop working if we will switch repository implementation. Please fix it and tell me why it will happens.